StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerViewController.swift

```diff
@@ -458,100 +604,15 @@ extension LinkAccountPickerViewController: LinkAccountPickerBodyViewDelegate {
                             )
                     }
 
-                    let deselectPreviouslySelectedAccount = { [weak self] in
-                        guard let self = self else { return }
-                        self.dataSource.updateSelectedAccounts(
-                            self.dataSource.selectedAccounts.filter(
-                                { $0.partnerAccount.id != selectedPartnerAccount.id }
-                            )
+                    // if we need to acquire consent, instead of showing the drawer now,
+                    // we will show the drawer when user presses the CTA where
+                    // pressing the CTA will make a call to `markConsentAcquired`
+                    if !dataSource.acquireConsentOnPrimaryCtaClick {
```

Comment:
this is the key part:  IF we need to acquire consent on CTA click, we will NOT present the drawer, here....we will present the drawer after acquiring consent (after CTA click)